cloud_experts_tutorials/cloud-experts-deploying-application/cloud-experts-deploying-s2i-webhook-cicd.adoc

```diff
@@ -0,0 +1,98 @@
+:_mod-docs-content-type: ASSEMBLY
+[id="cloud-experts-deploying-s2i-webhook-cicd"]
+= Tutorial: Using Source-to-Image (S2I) webhooks for automated deployment
+include::_attributes/attributes-openshift-dedicated.adoc[]
+:context: cloud-experts-deploying-s2i-webhook-cicd
+:source-highlighter: coderay
+
+toc::[]
+
+//rosaworkshop.io content metadata
+//Brought into ROSA product docs 05-07-2024
+
+Automatically trigger a build and deploy anytime you change the source code by using a webhook. For more information about this process, see link:https://docs.openshift.com/container-platform/latest/cicd/builds/triggering-builds-build-hooks.html[Triggering Builds].
+
+.Procedure
+
+. In your terminal, run the following command to get the GitHub webhook trigger secret:
++
+[source,terminal]
+----
+$ oc get bc/ostoy-microservice -o=jsonpath='{.spec.triggers..github.secret}'
+----
++
+.Example output
+[source,terminal]
+----
+`o_3x9M1qoI2Wj_cz1WiK`
+----
++
+[IMPORTANT]
+====
+You need to use this secret in a later step in this process.
+====
+
+. Run the following command to get the GitHub webhook trigger URL from the OSToy's buildconfig:
++
+[source,terminal]
+----
+$ oc describe bc/ostoy-microservice
+----
++
+.Example output
+[source,terminal]
+----
+[...]
+Webhook GitHub:
+	URL:	https://api.demo1234.openshift.com:443/apis/build.openshift.io/v1/namespaces/ostoy-s2i/buildconfigs/ostoy/webhooks/<secret>/github
+[...]
+----
+
+. In the GitHub webhook URL, replace the `<secret>` text with the actual secret you retrieved. Your URL will look like the following:
++
+.Example output
+[source,text]
+----
+https://api.demo1234.openshift.com:443/apis/build.openshift.io/v1/namespaces/ostoy-s2i/buildconfigs/ostoy-microservice/webhooks/o_3x9M1qoI2Wj_czR1WiK/github
+----
+
+. Setup the webhook URL in GitHub repository.
+.. In your repository, click *Settings > Webhooks > Add webhook*.
++
+image::ostoy-webhook.png[Add Webhook]
++
+.. Paste the GitHub webhook URL with the `Secret` included into the "Payload URL" field.
+.. Change the "Content type" to `application/json`.
+.. Click then *Add webhook* button.
++
+image::ostoy-webhookfinish.png[Finish Add Webhook]
++
+You should see a message from GitHub stating that your webhook was successfully configured. Now, whenever you push a change to your GitHub repository, a new build automatically starts, and upon a successful build, a new deployment starts.
+
+. Now, make a change in the source code. Any changes automatically trigger a build and deployment. In this example, the colors that denote the status of your OSToy app are selected randomly. Change the box to only display grayscale.
```

Comment:
Is this an example of a way to test to make sure the process is working? IF so, I wonder if that should be spelled out, ie  "To test the configuration, do this...."